### PR TITLE
[DOC] Use RDoc inclusions in string.c

### PR DESCRIPTION
**Pull request by BurdetteLamar**
As @peterzhu2118 and @duerst have pointed out, putting string method's RDoc into doc/ (which allows non-ASCII in examples) makes the "click to toggle source" feature not work for that method.

This PR moves the primary method doc back into string.c, then includes RDoc from doc/string/*.rdoc, and also removes doc/string.rdoc.

The affected methods are:
- ::new
- #bytes
- #each_byte
- #each_line
- #split

The call-seq is in string.c because it works there; it did not work when the call-seq is in doc/string/*.rdoc.

This PR also updates the relevant guidance in doc/documentation_guide.rdoc.

### CODE REVIEW

File: doc/string/bytes.rdoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Returns an array of the bytes in +self+:
+
+  'hello'.bytes # => [104, 101, 108, 108, 111]
+  'тест'.bytes  # => [209, 130, 208, 181, 209, 129, 209, 130]
+  'こんにちは'.bytes
+  # => [227, 129, 147, 227, 130, 147, 227, 129, 171, 227, 129, 161, 227, 129, 175]
+
+
+
```
Comment (peterzhu2118) [inline doc/string/bytes.rdoc:9]: Can you remove the extra newlines at the end of the files?

**Review by peterzhu2118 — APPROVED** [review]
One comment about whitespace at the end of files, otherwise looks good!

File: doc/string/bytes.rdoc (hunk 1 above)
Comment (BurdetteLamar) [reply doc/string/bytes.rdoc]: Done.

**Comment by BurdetteLamar** [comment]
This involves a structural change, so I'm going for two approvals.

**Comment by BurdetteLamar** [comment]
@jeremyevans, will you have time to look at this.  Since it changes structure, I'd like a second review/approval.

**Pull request merged**